src/AspNet.Security.OAuth.Deezer/DeezerAuthenticationHandler.cs

```diff
@@ -0,0 +1,154 @@
+                return OAuthTokenResponse.Failed(new Exception(error));
+            }
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
+        {
+            string address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, "access_token", tokens.AccessToken);
+
+            using var request = new HttpRequestMessage(HttpMethod.Get, address);
+            request.Headers.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+            using var response = await Backchannel.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, Context.RequestAborted);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving the user profile.");
+            }
+
+            using var payload = JsonDocument.Parse(await response.Content.ReadAsStringAsync());
+
+            var principal = new ClaimsPrincipal(identity);
+            var context = new OAuthCreatingTicketContext(principal, properties, Context, Scheme, Options, Backchannel, tokens, payload.RootElement);
+            context.RunClaimActions();
+
+            await Options.Events.CreatingTicket(context);
+            return new AuthenticationTicket(context.Principal, context.Properties, Scheme.Name);
+        }
+
+        protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri)
+        {
+            var scopeParameter = properties.GetParameter<ICollection<string>>(OAuthChallengeProperties.ScopeKey);
+            string scopes = scopeParameter != null ? FormatScope(scopeParameter) : FormatScope();
+
+            var parameters = new Dictionary<string, string>
+            {
+                { "app_id", Options.App_Id },
+                { "redirect_uri", redirectUri },
+                { "perms", scopes }
+            };
+
+            if (Options.UsePkce)
+            {
+                byte[] bytes = new byte[32];
+                CryptoRandom.GetBytes(bytes);
+                string codeVerifier = Microsoft.AspNetCore.WebUtilities.Base64UrlTextEncoder.Encode(bytes);
+
+                // Store this for use during the code redemption.
+                properties.Items.Add(OAuthConstants.CodeVerifierKey, codeVerifier);
+
+                using var sha256 = SHA256.Create();
```

Comment:
Use `HashAlgorithm.Create("SHA256")`

Reply:
Can you please explain in short why it is better to use `HashAlgorithm.Create("SHA256")` instead of `SHA256.Create()`? I wasn't able to find any good reason.
It seems also, that both variants do the same: 

[Stackoverflow - Is there any difference between SHA256.Create() and HashAlgorithm.Create(“SHA-256”)?](https://stackoverflow.com/a/34039317/2487565)

Reply:
The main reason is that it just delegates the specific implementation choice into the framework itself, and `HashAlgorithm` is less-derived than `SHA256` for a hash algorithm.

Although .NET Core doesn't do this, in .NET Framework it was also possible to reconfigure the implementations for each string at this point, so you could more flexibly update implementations without changing code ([docs](https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/map-algorithm-names-to-cryptography-classes)).

Reply:
Thank you!